### ISSUE TITLE
German keyboard special chars not working as expected

### ISSUE THREAD

**Issue by michag86** [rustdesk/rustdesk#2609]
**Describe the bug you encountered:**
I was connected from Linux (Ubuntu 22.04) to Windows 10. Under Linux I used X instead of Wayland, because wayland works only with keyboard map mode, where only the normal letters and numbers are available but almost no special character. I hope wayland keyboard support will be improved soon because ubuntu 22.04 is now switched to wayland as replacement for X.

After switching to X it worked better but a lot of characters on the keyboard are not working or generate other characters.

**What did you expect to happen instead?**
here is a list of "expected character -> resulting character"
```
° -> `
^ -> ´
ß -> -
? -> _
\ -> nothing
´ -> 0
` -> =
ü -> 8
Ü -> (
ö -> ,
Ö -> ;
ä -> #
Ä -> '
+ -> 9
* -> )
~ -> ]
# -> ß
' -> ?
< -> nothing
> -> nothing
| -> nothing
- -> 7
_ -> /
```
Layout can be found here:
https://en.wikipedia.org/wiki/German_keyboard_layout#/media/File:KB_Germany.svg

~I also noticed, that page up and page down are not working~ ([PEBCAK](https://en.wiktionary.org/wiki/PEBCAK))

**How did you install `RustDesk`?**

builds from github

---

**RustDesk version and environment**

Controlling Client (Ubuntu 22.04 with X Server): github nightly from 2022-12-11
Controlled Client (Windows 10 21H2 Build 19044.1288): github nightly from 2022-12-19

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/issues/1547

**Comment by fufesou** [comment]
@michag86 Hi, thanks for your feedback.
Would you please try the deb nightly build today.

**Comment by michag86** [comment]
I'm a bit confused.

Is the map mode now the default setting?

In map mode it works a lot better.
I noticed only this wrong mapping:
```
< -> #
> -> '
| -> nothing
```

But the compatibility mode has still the issues as written in my first post. I also noticed, that in compatibility mode the letters z and y are swapped (German keyboard ist qwertz, not qwerty).

**Comment by fufesou** [comment]
@michag86 Thanks, the feedback is very useful.
Map mode is the mode under dev. I'm trying to fix the wrong mapping.
The compatibility mode will be fixed after the mapping is corrected.

**Comment by fufesou** [comment]
Hi @michag86. The `IntlBackslash` key is fixed. But it's only for Win and Linux. MacOS map mode is not well supported now.
https://github.com/rustdesk/rustdesk/pull/2624

There're some keyboard bugs in today's nightly. I'll notice here if keyboard is OK.

**Comment by fufesou** [comment]
@michag86 Can you please check the mapping is fixed?
```
< -> #
> -> '
| -> nothing
```

**Comment by michag86** [comment]
@fufesou Tested and works :+1: 

So now it is working under ubuntu 22.04 with x server.

But the key mapping is still not working with wayland.

There is an issue on that here https://github.com/rustdesk/rustdesk/issues/2604

But it was closed by the create because of an workaround. I think it should be open until keyboard mapping in wayland is working.

I noticed that there are some letters, when I input the numbers above the letters.

This ticket can be closed.
